source/slang/slang-type-layout.cpp

```diff
@@ -3613,43 +3661,65 @@ static TypeLayoutResult _createTypeLayout(
             typeLayout->type = type;
             typeLayout->rules = rules;
 
-            if (isCPUTarget(context.targetReq) || isCUDATarget(context.targetReq))
+            LayoutSize fixedExistentialValueSize = 0;
+            bool useFixedSizedExistentialValue =
+                isCPUTarget(context.targetReq) || isCUDATarget(context.targetReq);
+            if (useFixedSizedExistentialValue)
             {
-                LayoutSize fixedSize = 16;
+                fixedExistentialValueSize = 16;
                 if (auto anyValueAttr =
                         interfaceDeclRef.getDecl()->findModifier<AnyValueSizeAttribute>())
                 {
-                    fixedSize += anyValueAttr->size;
+                    fixedExistentialValueSize = anyValueAttr->size;
                 }
-                else
-                {
-                    // The interface type does not have an `[anyValueSize]` attribute,
-                    // assume a default of 8 bytes.
-                    fixedSize += 8;
-                }
-                typeLayout->addResourceUsage(LayoutResourceKind::Uniform, fixedSize);
+                typeLayout->addResourceUsage(LayoutResourceKind::Uniform, fixedExistentialValueSize + 16);
```

Comment:
The `+ 16` here could probably do with a comment. That is for the RTTI and witness table fields, but it might not be clear to the reader of this code.